Library/Homebrew/metaformula.rb

```diff
@@ -61,5 +61,24 @@ def bottle
     def patch
       raise "`patch` DSL is disabled in MetaFormula"
     end
+
+    # A new DSL introduced in {MetaFormula} is {.depends_on_cask}
+    # <pre>
+    #   depends_on_cask "some_cask"
+    # </pre>
+    #
+    # this is implicitly equivalent to
+    # `depends_on :__meta_cask_req_some_cask`
+    # And this special prefix will help the dep to be parsed into {CaskRequirement}
+    # The special prefix is defined in {.cask_req_prefix}
+    def depends_on_cask(cask)
```

Comment:
Instead of making this a new DSL method, I wonder whether it might be a good idea to make this a new symbol spec like the other ones that currently exist. For example, something like `depends_on :cask => "name"`. That way you would need less special handling, and it could be managed in the same case statement that the rest of `parse_symbol_spec` uses. What do you think, @GauthamGoli?

Reply:
True, I think that may be better solution.

Reply:
cask vs. requirement name collisions
```
depends_on :java
```

also cask @ formula naming would be impossible since @ isn't a valid character in ruby symbols

Reply:
FWIW,
```
depends_on :cask => ["name", :build]
```
strikes me as a little weird, which is a technical term.

Reply:
>also cask @ formula naming would be impossible since @ isn't a valid character in ruby symbols

also `-` isn't allowed, which is used extensively in cask names already

Reply:
> also cask @ formula naming would be impossible since @ isn't a valid character in ruby symbols

Not the prettiest syntax, but you can put whatever in a symbol via `:"@this is a real symbol!"`

But as you pointed out there are some ambiguities in using symbols this way, so we shouldn't use that for the cask specification.

Reply:
TIL